psl/builtin-connectors/src/postgres_datamodel_connector/validations.rs

```diff
@@ -474,3 +474,37 @@ pub(super) fn extensions_preview_flag_must_be_set(
         span,
     ));
 }
+
+pub(super) fn extension_names_follow_prisma_syntax_rules(
+    preview_features: BitFlags<PreviewFeature>,
+    props: &PostgresDatasourceProperties,
+    errors: &mut Diagnostics,
+) {
+    if !preview_features.contains(PreviewFeature::PostgresqlExtensions) {
+        return;
+    }
+
+    let extensions = match props.extensions() {
+        Some(extensions) => extensions,
+        None => return,
+    };
+
+    for extension in extensions.extensions() {
+        if extension.name.is_empty() {
+            errors.push_error(DatamodelError::new_validation_error(
+                "The name of an extension must not be empty.",
+                extension.span,
+            ));
+        } else if extension.name.chars().next().unwrap().is_numeric() {
+            errors.push_error(DatamodelError::new_validation_error(
+                "The name of an extension must not start with a number.",
+                extension.span,
+            ));
+        } else if extension.name.contains('_') {
+            errors.push_error(DatamodelError::new_validation_error(
+                "The character `-` is not allowed in extension names.",
+                extension.span,
+            ))
+        }
+    }
+}
```

Comment:
These rules are already in `validate_identifier()` in parser-database (just checked). I'm worried things will get out of sync. Can you add a comment so we don't forget to come back to this at some point? `validate_identifier()` doesn't sound like a good approach either, so I'd rather keep this function here as it is and think about it later.

Reply:
It's super tricky to reuse that code. I added a comment.